### PR TITLE
Dumb indicator replacement

### PR DESCRIPTION
**Pull request by tinkermakar**
details here: https://github.com/marcinjahn/gnome-peek-top-bar-on-fullscreen-extension/issues/8

### CODE REVIEW

File: resources/metadata.json [hunk 1]
```diff
@@ -4,5 +4,5 @@
   "uuid": "peek-top-bar-on-fullscreen@marcinjahn.com",
   "shell-version": ["43", "44"],
   "url": "https://github.com/marcinjahn/gnome-peek-top-bar-on-fullscreen-extension",
-  "version": 7
+  "version": 8
```
Comment (marcinjahn) [inline resources/metadata.json:7]: Please don't update the version yet. I usually do that in a separate commit.

Comment (tinkermakar) [reply resources/metadata.json]: done

File: src/panel/wayland-panel-manager.ts [hunk 2]
```diff
@@ -1,58 +1,33 @@
...
- * top bar indicator is added and activated when the panel is supposed
- * to show up. That causes the panel to be visible, at the cost
- * of requiring the user to click outside of the panel to hide it.
+ * even though it becomes clickable. As a workaround, on Wayland a concealed dumb
+ * window with active always on top makes it work as expected.
  */
 export class WaylandPanelManager implements PanelManager {
+  constructor() {
+    spawn_command_line_async(
+      'bash -c "GDK_BACKEND=x11 gjs .local/share/gnome-shell/extensions/peek-top-bar-on-fullscreen@marcinjahn.com/dummy-window.js"'
```
Comment (marcinjahn) [inline src/panel/wayland-panel-manager.ts:16]: I'd switch from `bash` to `sh`, since bash might not be available on the host (I doubt it wouldn't, but it's safer to use sh)

File: src/panel/wayland-panel-manager.ts [hunk 3]
```diff
@@ -1,58 +1,33 @@
...
 
   hidePanel(): void {
     PanelBox.visible = false;
-    StatusArea[TOP_BAR_DUMB_INDICATOR]?.destroy();
   }
 
-  resetAnyTweaks(): void {
-    StatusArea[TOP_BAR_DUMB_INDICATOR]?.destroy();
-  }
+  resetAnyTweaks() {}
```
Comment (marcinjahn) [inline src/panel/wayland-panel-manager.ts:28]: This method can be removed from all manager classes

File: src/panel/wayland-panel-manager.ts (hunk 2 above)
Comment (marcinjahn) [inline src/panel/wayland-panel-manager.ts:16]: The path to the app should rather be built using `ExtensionUtils.getCurrentExtension().path`.

File: src/panel/wayland-panel-manager.ts [hunk 4]
```diff
@@ -1,58 +1,33 @@
...
 
 /**
  * On Wayland, making the panel visible is not enough,
  * there is some weird issue that causes the panel to stay invisible,
- * even though it becomes clickable. As a workaround, on Wayland a dumb
- * top bar indicator is added and activated when the panel is supposed
- * to show up. That causes the panel to be visible, at the cost
- * of requiring the user to click outside of the panel to hide it.
+ * even though it becomes clickable. As a workaround, on Wayland a concealed dumb
+ * window with active always on top makes it work as expected.
```
Comment (marcinjahn) [inline src/panel/wayland-panel-manager.ts:11]: window that is always on top
```suggestion
 * app with invisible window (always on top) is started. That makes the panel visible
```

**Comment by marcinjahn** [comment]
You could also remove lines 16-17 from `README.md`. Similarly, the same piece of information can be removed from https://github.com/marcinjahn/gnome-peek-top-bar-on-fullscreen-extension/blob/main/resources/metadata.json, since it's no longer true.

File: src/panel/wayland-panel-manager.ts [hunk 5]
```diff
@@ -1,58 +1,33 @@
...
  */
 export class WaylandPanelManager implements PanelManager {
+  constructor() {
+    spawn_command_line_async(
```
[the remark this answers is not in the document]
Comment (tinkermakar) [reply src/panel/wayland-panel-manager.ts:17]: all done, except for this one -- I am not sure I am getting what exactly needs to be done, since

1. constructor is basically init() in itself, and
2. we don't have an `X11Manager` anywhere. We have `X11PanelManager`, but the tweak is irelevant to X11

[the remark this answers is not in the document]
Comment (marcinjahn) [reply src/panel/wayland-panel-manager.ts:17]: Right, I meant `X11PanelManager`, sorry.
In general, I think, and it's not only my opinion, constructors should not do any "big" things outside of initializing the object itself. Creating an external process when invoking a constructor is a "big" thing, and it could be a bit unexpected. That's why I'd prefer for the process to be created in some `init`, or alternatively, there could be some static method in `WaylandPanelManager` that creates the instance and starts the process. Anything other than `new WaylandPanelManager()` will be better in my opinion. When I see `new Something()` I expect the object to be created and initialized with some default internal state. I don't expect some app to be started in another process :)

Having said that, it's alright if we leave it as it is right now, and I can introduce changes in another commit :)
